File: Java/100. Same Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p==null|| q==null){
            return p==q;
        }
        if(isSameTree(p.left,q.left) && isSameTree(p.right,q.right) && p.val==q.val){return true;}
        return false;
    }
}

File: C/100. Same Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
bool isSameTree(struct TreeNode* p, struct TreeNode* q){
    if(p==NULL|| q==NULL){
        return p==q;
    }
    if(isSameTree(p->left,q->left) && isSameTree(p->right,q->right) && p->val==q->val){return true;}
    return false;
}
